core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -138,21 +146,29 @@ private int readInternal(byte[] b, int off, int len, ReadType readType, long pos
       } else {
         pageId = new PageId(Long.toString(mStatus.getFileId()), currentPage);
       }
+      readTimeStopwatch.start();
       int bytesRead =
           mCacheManager.get(pageId, currentPageOffset, bytesLeftInPage, b, off + totalBytesRead,
               mCacheContext);
+      readTimeStopwatch.stop();
       if (bytesRead > 0) {
         totalBytesRead += bytesRead;
         currentPosition += bytesRead;
         Metrics.BYTES_READ_CACHE.mark(bytesRead);
         if (cacheContext != null) {
           cacheContext
               .incrementCounter(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE.getMetricName(), bytesRead);
+          cacheContext.incrementCounter(
+              MetricKey.CLIENT_CACHE_PAGE_READ_CACHE_TIME_NS.getMetricName(),
+              readTimeStopwatch.elapsed(TimeUnit.NANOSECONDS));
         }
       } else {
         // on local cache miss, read a complete page from external storage. This will always make
         // progress or throw an exception
+        readTimeStopwatch.reset();
+        readTimeStopwatch.start();
```

Comment:
possibly to have this in `try-with-resource`? note that `readExternalPage` may throw IOE

Reply:
Good call, but I have no idea about how to make it in a try-with-resource,  because the stopwatch is not a closable resource.  I just read through the implementation of stopwatch, looks like if any exception occurs, we might skip the invocation of stop(), but it won't make anything wrong, because we call reset() every time before start().